src/benchq/problem_ingestion/molecule_instance_generation.py

```diff
@@ -48,49 +49,111 @@ class ChemistryApplicationInstance:
     occupied_indices: Optional[Iterable[int]] = None
     active_indices: Optional[Iterable[int]] = None
 
-    def get_molecular_data(self) -> PyscfMolecularData:
-        """Run an SCF calculation using PySCF and return the results as a MolecularData
-        object."""
-        return run_pyscf(
-            MolecularData(
-                self.geometry,
-                self.basis,
-                self.multiplicity,
-                self.charge,
+    def get_pyscf_molecule(self) -> gto.Mole:
+        "Generate the PySCF molecule object describing the system to be calculated."
+        pyscf_molecule = gto.Mole()
+        pyscf_molecule.atom = self.geometry
+        pyscf_molecule.basis = self.basis
+        pyscf_molecule.spin = self.multiplicity - 1
+        pyscf_molecule.charge = self.charge
+        pyscf_molecule.symmetry = False
+        pyscf_molecule.build()
+        return pyscf_molecule
+
+    def _run_pyscf(self) -> Tuple[gto.Mole, scf.hf.SCF]:
+        """Run an SCF calculation using PySCF and return the results as a meanfield
+        object.
+
+        Note that this method will apply AVAS but does not account for occupied_indices
+        and active_indices.
+
+        Returns:
+            Tuple whose first element is the PySCF molecule object after AVAS reduction
+                and whose second is the meanfield object containing the SCF solution.
+        """
+        molecule = self.get_pyscf_molecule()
+        mean_field_object = (scf.RHF if self.multiplicity == 1 else scf.ROHF)(molecule)
+        mean_field_object.run()
+        if self.avas_atomic_orbitals or self.avas_minao:
+            molecule, mean_field_object = truncate_with_avas(
+                mean_field_object,
+                self.avas_atomic_orbitals,
+                self.avas_minao,
             )
-        )
+        return molecule, mean_field_object
 
     def get_active_space_hamiltonian(self) -> openfermion.InteractionOperator:
         """Generate the fermionic Hamiltonian corresponding to the instance's
-        active space."""
-        if self.avas_atomic_orbitals or self.avas_minao:
-            raise ValueError(
-                "Generating the active space Hamiltonian for application instances"
-                "with AVAS is not currently supported."
-            )
-        return self.get_molecular_data().get_molecular_hamiltonian(
+        active space.
+
+        Returns:
+            The fermionic Hamiltonian corresponding to the instance's active space. Note
+                that the active space will account for both AVAS and the
+                occupied_indices/active_indices attributes.
+        """
+        return self._get_molecular_data().get_molecular_hamiltonian(
             occupied_indices=self.occupied_indices,
             active_indices=self.active_indices,
         )
 
     def get_active_space_meanfield_object(self) -> scf.hf.SCF:
```

Comment:
Maybe we should rename method's signature so it indicates that we get active space accounting for AVAS? WDYT?

Reply:
Good question. I had left AVAS out of the name with the idea that the person who wants to get the active space Hamiltonian probably doesn't know about what methods a particular application instance uses to construct the active space (AVAS, freezing orbitals with certain indices, or in the future some other techniques).

I did update the docstring to make this more clear in 356fbaf99168864c11c. What do you think?

Reply:
I like it! Thanks.